src/ImportDefinitionsBundle/Command/CreateExportCommand.php

```
@@ -0,0 +1,130 @@
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class CreateExportCommand extends AbstractCommand
+{
+    /**
+     * @var MetadataInterface
+     */
+    protected $metadata;
+
+    /**
+     * @var RepositoryInterface
+     */
+    protected $repository;
+
+    /**
+     * @var ObjectManager
+     */
+    protected $manager;
+
+    /**
+     * @var ResourceFormFactoryInterface
+     */
+    protected $resourceFormFactory;
+
+    public function __construct(
+        MetadataInterface $metadata,
+        RepositoryInterface $repository,
+        ObjectManager $manager,
+        ResourceFormFactoryInterface $resourceFormFactory
+    ) {
+        $this->metadata = $metadata;
+        $this->repository = $repository;
+        $this->manager = $manager;
+        $this->resourceFormFactory = $resourceFormFactory;
+
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('export-definitions:create')
+            ->setDescription('Create a Export Definition.')
+            ->setHelp(<<<EOT
+The <info>%command.name%</info> creates a Export Definition.
+EOT
+            )
+            ->addArgument(
+                'path',
+                InputArgument::REQUIRED,
+                'Path to Export Definition JSON export file'
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws \Exception
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $path = $this->getPath();
+
+        $jsonContent = file_get_contents($path);
+        $data = json_decode($jsonContent, true);
+
+        try {
+            $definition = ExportDefinition::getByName($data['name']);
+        } catch (\InvalidArgumentException $e) {
+            $definition = new ExportDefinition();
```

I would throw an exception here, since the definition doesn't exist

---

If the definition doesn't exist I want create it. This ist the main reason. The try block is for update.

---

👍